### PR TITLE
Feature/survey for patients

### PR DESCRIPTION
**Pull request by Aleksa1998**

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Repository/IPatientsRepository.cs [hunk 1]
```diff
@@ -10,5 +10,6 @@ public interface IPatientsRepository
         PatientUser Validate(PatientUser patient);
         List<PatientUser> GetAll();
         PatientUser FindOne(int id);
+
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Repository/IPatientsRepository.cs:13]: Unnecessarily added blank line needs to be deleted.

File: PatientWebApplication/HealthClinic.DAL/Service/ISurveyService.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace HealthClinic.CL.Service
+{
+    public interface ISurveyService
+    {
+        Survey Create(SurveyDto surveyDto);
+        List<Survey> GetAllSurveysForPatientId(int id);
+
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/ISurveyService.cs:13]: The blank line needs to be deleted.

File: PatientWebApplication/HealthClinic.DAL/Service/SurveyService.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        public List<Survey> GetAllSurveysForPatientId(int id)
+        {
+            return SurveyRepository.GetAllSurveysForPatientId(id);
+        }
+
+        public List<DoctorAppointment> FindAllValidAppointments(List<DoctorAppointment> allValidAppointments)
+        {
+            List<Survey> allSurveys = GetAllSurveysForPatientId(1);
+            List<int> allUnvalidAppointments = FindAllUnvalidAppointments(allSurveys);
+            return CheckIfAppointmentsHappened(allValidAppointments.Where(p => !allUnvalidAppointments.Any(p2 => p2 == p.id)).ToList());
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/SurveyService.cs:36]: Write in the form:
"return CheckIfAppointmentsHappened(FindAllUnvalidAppointments(GetAllSurveysForPatientId(1)).Where(p => !allUnvalidAppointments.Any(p2 => p2 == p.id)).ToList());"

File: PatientWebApplication/HealthClinic.DAL/Service/SurveyService.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        private static List<int> FindAllUnvalidAppointments(List<Survey> allSurveys)
+        {
+            List<int> allUnvalidAppointments = new List<int>();
+            foreach (Survey survey in allSurveys)
+            {
+                allUnvalidAppointments.Add(survey.appointmentId);
+            }
+
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/SurveyService.cs:46]: The blank line needs to be deleted.

File: PatientWebApplication/PatientWebApplication/Controllers/SurveyController.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            }
+            return Ok();
+        }
+
+        [HttpGet]      
+        public IActionResult Get()
+        {       
+            List<DoctorAppointment> allValidAppointments = regularAppointmentService.GetAppointmentsForPatient(1);          
+            allValidAppointments = SurveyService.FindAllValidAppointments(allValidAppointments);
+            return Ok(allValidAppointments);
```
Comment (trki293) [inline PatientWebApplication/PatientWebApplication/Controllers/SurveyController.cs:41]: Write in the form:
"return Ok(SurveyService.FindAllValidAppointments(regularAppointmentService.GetAppointmentsForPatient(1)));"

**Review by trki293 — CHANGES_REQUESTED** [review]
Necessary changes to the indicated places in the code.

**Review by trki293 — APPROVED** [review]
Good practice of Clean code satisfied, PR approved.

**Review by Maja0505 — APPROVED** [review]
PR is approved, all Clean code principles are followed.

**Pull request merged**